kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -101,18 +101,27 @@ public static void validateSchema(StructType schema, boolean isColumnMappingEnab
    * </ul>
    */
   public static void validateUpdatedSchema(
```

Comment:
Just refactored to not duplicate info across different fields (we can get this info from just the metadata)